### PR TITLE
Decimal input

### PR DESCRIPTION
**Pull request by notks**

### CODE REVIEW

File: src/pages/invoices/common/hooks/useResolveInputField.tsx [hunk 1]
```diff
@@ -16,13 +16,19 @@ import { useHandleProductChange } from './useHandleProductChange';
...
 import { isNonNumericValue } from 'common/helpers/invoices/resolve-non-numeric-value';
+import { CurrencyInput } from 'components/forms/CurrencyInput';
+import { ClientResolver } from 'common/helpers/clients/client-resolver';
+import { CurrencyResolver } from 'common/helpers/currencies/currency-resolver';
+import { Client } from 'common/interfaces/client';
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
+import { Currency } from 'common/interfaces/currency';
 
-const numberInputs = ['discount', 'cost', 'unit_cost', 'quantity'];
+const numberInputs = ['discount', 'unit_cost', 'quantity'];
```
Comment (beganovich) [inline src/pages/invoices/common/hooks/useResolveInputField.tsx:31]: Can you share the reason why `cost` was removed?

File: src/pages/invoices/common/hooks/useResolveInputField.tsx [hunk 2]
```diff
@@ -32,13 +38,35 @@ interface Props {
...
+      );
+      currency && setCurrency(currency);
+    } else {
+      const currency: Currency | undefined = await currencyresolver.find(
+        company.settings?.currency_id
+      );
+      currency && setCurrency(currency);
+    }
+  };
+  useEffect(() => {
```
Comment (beganovich) [inline src/pages/invoices/common/hooks/useResolveInputField.tsx:67]: Please add blank lines for clarity.

File: src/pages/invoices/common/hooks/useResolveInputField.tsx [hunk 3]
```diff
@@ -71,6 +99,23 @@ export function useResolveInputField(props: Props) {
         />
       );
     }
+    if (property === 'cost') {
+      console.log('resolve curr', currency);
```
Comment (beganovich) [inline src/pages/invoices/common/hooks/useResolveInputField.tsx:103]: Please remove all logging.

File: src/pages/invoices/common/hooks/useResolveInputField.tsx [hunk 4]
```diff
@@ -32,13 +38,35 @@ interface Props {
...
   const onChange = useHandleLineItemPropertyChange();
+
   const invoice = useCurrentInvoice();
   const formatMoney = useFormatMoney();
+  const company = useCurrentCompany();
+
+  const clientResolver = new ClientResolver();
+  const currencyresolver = new CurrencyResolver();
+
+  const getCurrency = async (client_id: string) => {
```
Comment (beganovich) [inline src/pages/invoices/common/hooks/useResolveInputField.tsx:53]: From what I can see we aren't calling this function async?

File: src/pages/recurring-invoices/common/hooks/useResolveInputField.tsx [hunk 5]
```diff
@@ -31,11 +37,37 @@ interface Props {
...
   const { setIsTaxModalOpen, setIsProductModalOpen } = props;
+
   const invoice = useCurrentRecurringInvoice();
   const formatMoney = useFormatMoney();
+  const company = useCurrentCompany();
+
+  const clientResolver = new ClientResolver();
+  const currencyresolver = new CurrencyResolver();
+
+  const getCurrency = async (client_id: string) => {
```
Comment (beganovich) [inline src/pages/recurring-invoices/common/hooks/useResolveInputField.tsx:53]: Is it possible to extract getCurrency method into custom hook so we can reuse it in both components?

**Review by beganovich — CHANGES_REQUESTED** [review]
Looks good, minor changes are needed.

File: src/pages/invoices/common/hooks/useResolveInputField.tsx (hunk 1 above)
Comment (notks) [reply src/pages/invoices/common/hooks/useResolveInputField.tsx]: Initially I thought this input should be used for item cost only and i did separate check for that column but it has come to my knowledge that this input is used for all number inputs on invoice page.

**Comment by beganovich** [comment]
Looks good, minor change is needed.

**Review by beganovich — CHANGES_REQUESTED** [review]

**Comment by beganovich** [comment]
Thanks for the contribution 👍

**Pull request merged**